Update README.md for single sourcing (#1194)
rancher/rancher-product-docs: README.md

Reviewer: This section needs to be updated too. Maybe we can use the same pattern as [doc-sle](https://github.com/SUSE/doc-sle/blob/main/README.adoc?plain=1#L1-L18).

Reviewer: Maybe we should drop this `### Clone the Repository` section since we're telling readers to fork and clone in the `### Edit the Documentation` section around line 114.

npm ci
```

### Build the Static Website Using the Local Documentation Content

> There are several playbook files available to build different variants of the documentation site, such as the community, product (Rancher Prime), or SRFA (SUSE Rancher for AWS) site, using either local or remote content.

Reviewer 2: ```suggestion
There are several playbook files available to build different variants of the documentation site, such as the community (Rancher Manager), product (Rancher Prime), or SRFA (SUSE Rancher for AWS) site, using either local or remote content.
```

1. [Open an issue](https://github.com/rancher/rancher-product-docs/issues/new/choose).
1. Edit the documentation in the way you see fit and submitting a [pull request](https://github.com/rancher/rancher-product-docs/compare).

### Edit the Documentation

To get started, [fork](https://github.com/rancher/rancher-product-docs/fork) and clone the `rancher-product-docs` repository.

Our repository does not allow you to make changes directly to the `main` branch. Create a working branch and make pull requests from your fork to [rancher/rancher-product-docs](https://github.com/rancher/rancher-product-docs).

### Branching Strategy

Most documentation development happens on the `main` branch. However, when working on version-specific content, your pull requests should target a version branch (for example, `v2.10.10`). These version branches are created at the beginning of the development cycle for a specific release. Once the version is released, the release branch is merged back into `main`. 

> An exception to this is a new minor version (such as `v2.10.0`), for which development is opened on `main` from the beginning of the development cycle.

Reviewer: This is not true as of this moment (but may change).

Author: Ah I see, might feel a bit counterintuitive to have the new minor directory tree on main and a release branch at the same time, so I'd vote for changing it in the future.

### File Organization

#### Content and Asset Files

There are two separate directory trees for documentation content:

- `community-docs/<VERSION>/`: Contains content for the community site, published at https://ranchermanager.docs.rancher.com/.

> - `versions/<VERSION>/`: Contains content for the product site, published at https://documentation.suse.com/.

Reviewer: ```suggestion
- `versions/<VERSION>/`: Contains content for the Prime site, published at https://documentation.suse.com/cloudnative/rancher-manager/.
```

- `community-docs/<VERSION>/modules/ROOT/pages/`

For product docs, the directory structure is as follows:

- `versions/<VERSION>/modules/<LOCALE>/pages/`

Other asset files (for example, images) have their own subdirectories. For example, `versions/<VERSION>/modules/en/images`.

#### Single-Sourcing with Conditionals

Because most documentation content is single-sourced and shared across the community, product (Rancher Prime), and SRFA (SUSE Rancher for AWS) sites, you might encounter situations where certain text, links, or instructions apply to only one specific build type.

To handle these differences, use AsciiDoc `ifeval::[]` conditionals based on the `{build-type}` attribute. This ensures the appropriate content is selectively rendered for the respective target site.

Here are some practical examples of how to use these conditionals in the `.adoc` files:

```asciidoc
// Content specifically for the community site
ifeval::["{build-type}" == "community"]
This content only appears in the Rancher Community documentation.
endif::[]

> // Content specifically for the product site

Reviewer: ```suggestion
// Content specifically for the Prime site
```

endif::[]

// Content specifically for the SUSE Rancher for AWS (SRFA) site
ifeval::["{build-type}" == "srfa"]
This content only appears in the SUSE Rancher for AWS documentation.
endif::[]

// Content for everything except the SRFA site (that is, applies to both product and community build types)
ifeval::["{build-type}" != "srfa"]
This content applies to both the Rancher Community and SUSE Rancher Prime documentation, but is excluded from SRFA.
endif::[]
```

For more information on the directive syntax, refer to the [AsciiDoc Conditionals documentation](https://docs.asciidoctor.org/asciidoc/latest/directives/conditionals/).

#### Required Anchor IDs for Headings

To support content translations, you must use explicit anchors for cross-references. Auto-generated anchors do not survive the translation process, which causes links to break.

When creating a heading, always define an explicit anchor directly above it. Use the following anchor format:

```asciidoc
[#_my_anchor]
== My Anchor

My paragraph.
```

You can then safely refer to this heading using the explicit anchor:

```asciidoc
xref:#_my_anchor[My Anchor]
```

> 

Reviewer 2: ```suggestion

Note: If you want to link from one documentation page to another, do not use absolute URLs. Antora uses a special `xref:` syntax to dynamically link files across versions and modules.

Example: xref:installation/requirements.adoc[System Requirements]
```

#### Updating Navigation

> When adding a new documentation page, or renaming or removing an existing one, you must update the navigation file (`nav.adoc`) for the respective version. This ensures that the site's sidebar menu accurately reflects the available content.

Reviewer: > respective version

Maybe we should be explicit it's version + locale

If the page is single-sourced, the `nav.adoc` file needs to be updated in both the community and product directory trees, for example:

- `community-docs/<VERSION>/modules/ROOT/nav.adoc`
- `versions/<VERSION>/modules/en/nav.adoc`

For more information on how to structure the navigation lists, refer to the [Antora Navigation Files and Lists documentation](https://docs.antora.org/antora/latest/navigation/files-and-lists/).

#### Configuration Files

- Site-level configuration is done through a playbook file. For example, `playbook-community-local.yml`. Refer to the [Antora playbook documentation](https://docs.antora.org/antora/latest/playbook/) for more information.
- Component-level configuration is done through a file called `antora.yml`. For example, `versions/<VERSION>/antora.yml` or `community-docs/<VERSION>/antora.yml`. Refer to the [Antora  antora.yml documentation](https://docs.antora.org/antora/latest/component-version-descriptor/) for more information.

> 

Reviewer 2: ```suggestion
#### Troubleshooting Common Issues

**Error:** `Antora documentation component not found` or CSS formatting looks completely broken.
**Fix:** You likely forgot to initialize the submodules. Run `git submodule update --init` and rebuild.

**Error:**  `npm ERR! code EBADENGINE` or dependency conflicts.
**Fix:** Ensure you are using a modern Node.js LTS version (e.g., Node 18 or Node 20). If you used `npm install`, delete your `node_modules` folder and run `npm ci` instead.

**Issue:** Changes aren't updating in the browser.
**Fix:** Antora does not feature "live-reload". Every time you make a change to a file, you must stop your terminal process, rerun `npx antora playbook-product-local.yml`, and refresh your browser.
```
